### ISSUE TITLE
RequireThisCheck: no violation produced when field is concatenated with String

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#7066]
````
$ cat TestClass.java
public class TestClass {
    private String id;

    public String method() {
        return "" + id;
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="RequireThisCheck">
<property name="checkFields" value="true" />
<property name="checkMethods" value="false" />
<property name="validateOnlyOverlapping" value="false" />
</module>
    </module>
</module>

$ java -jar checkstyle-8.24-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

I am expecting a violation on the return statement since `id` doesn't have `this.` in front of it.

If the string concatenation is removed, a violation shows up.

````
$ cat TestClass.java
public class TestClass {
    private String id;

    public String method() {
        return id;
    }
}

$ java -jar checkstyle-8.24-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:5:16: Reference to instance variable 'id' needs "this.". [RequireThis]
Audit done.
Checkstyle ends with 1 errors.
````

**Comment by romani** [comment]
I kind of fill guilty that we have such issue only because we do not activate this option in our config, we use config in default. We have bunch of examples in our code.
@rnveach , do you have open how did you catch such issue ? is it opensource project ? if yes we should use it in our regression testing.

I still think that activation of this property might be too much verbosity (the same as we have with `final` but it is already activated).

**Comment by romani** [comment]
fix is merged.
